ports/unix/modffi.c

```diff
@@ -266,19 +269,50 @@ STATIC mp_obj_t mod_ffi_func(mp_obj_t rettype, mp_obj_t addr_in, mp_obj_t argtyp
 }
 MP_DEFINE_CONST_FUN_OBJ_3(mod_ffi_func_obj, mod_ffi_func);
 
-STATIC void call_py_func(ffi_cif *cif, void *ret, void **args, void *func) {
+STATIC void call_py_func(ffi_cif *cif, void *ret, void **args, void *user_data) {
     mp_obj_t pyargs[cif->nargs];
+    mp_obj_fficallback_t *o = user_data;
+    mp_obj_t pyfunc = o->pyfunc;
+
+    if (o->lock) {
+        #if MICROPY_ENABLE_SCHEDULER
+        mp_sched_lock();
+        #endif
+        gc_lock();
+    }
+
     for (uint i = 0; i < cif->nargs; i++) {
         pyargs[i] = mp_obj_new_int(*(mp_int_t *)args[i]);
     }
-    mp_obj_t res = mp_call_function_n_kw(MP_OBJ_FROM_PTR(func), cif->nargs, 0, pyargs);
+    mp_obj_t res = mp_call_function_n_kw(pyfunc, cif->nargs, 0, pyargs);
 
     if (res != mp_const_none) {
         *(ffi_arg *)ret = mp_obj_int_get_truncated(res);
     }
+
+    if (o->lock) {
+        gc_unlock();
+        #if MICROPY_ENABLE_SCHEDULER
+        mp_sched_unlock();
+        #endif
+    }
 }
 
-STATIC mp_obj_t mod_ffi_callback(mp_obj_t rettype_in, mp_obj_t func_in, mp_obj_t paramtypes_in) {
+STATIC mp_obj_t mod_ffi_callback(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_rettype, ARG_func, ARG_parmtypes, ARG_lock };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_rettype,     MP_ARG_REQUIRED | MP_ARG_OBJ,  {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_func,        MP_ARG_REQUIRED | MP_ARG_OBJ,  {.u_obj = MP_OBJ_NULL} },
+        { MP_QSTR_paramtypes,  MP_ARG_REQUIRED | MP_ARG_OBJ,  {.u_obj = MP_OBJ_NULL} },
```

Comment:
because these 3 are required args, and were originally positional only (ie not kw), they can have just `MP_QSTR_` as their name, to save code space

Reply:
Done. Removed them completely from the `allowed_args` array.